pkg/cluster/fixetcd.go

```diff
@@ -0,0 +1,503 @@
+	peerPods, err := getPeerPods(pods.Items, de, cluster)
+	if err != nil {
+		return err
+	}
+
+	scc, cr, crb, sa, err := createPrivilegedServiceAccount(ctx, log, serviceAccountName, kubeServiceAccount, kubecli, securitycli)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		log.Infof("Deleting service account %s now", sa.Name)
+		err = kubecli.CoreV1().ServiceAccounts(nameSpaceEtcds).Delete(ctx, sa.Name, metav1.DeleteOptions{})
+	}()
+
+	defer func() {
+		log.Infof("Deleting %s now", scc.Name)
+		err = securitycli.Delete(ctx, scc.Name, metav1.DeleteOptions{})
+	}()
+
+	defer func() {
+		log.Infof("Deleting cluster role %s now", cr.Name)
+		err = kubecli.RbacV1().ClusterRoles().Delete(ctx, cr.Name, metav1.DeleteOptions{})
+	}()
+
+	defer func() {
+		log.Infof("Deleting cluster role binding %s now", crb.Name)
+		err = kubecli.RbacV1().ClusterRoleBindings().Delete(ctx, crb.Name, metav1.DeleteOptions{})
+	}()
+
+	log.Infof("Creating job %s", jobNameFixPeers)
+	b, err := kubecli.BatchV1().Jobs(nameSpaceEtcds).Create(ctx, &batchv1.Job{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      jobNameFixPeers,
+			Namespace: nameSpaceEtcds,
+			Labels:    map[string]string{"app": jobNameDataBackup},
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      jobNameFixPeers,
+					Namespace: nameSpaceEtcds,
+					Labels:    map[string]string{"app": jobNameDataBackup},
+				},
+				Spec: corev1.PodSpec{
+					RestartPolicy:      corev1.RestartPolicyOnFailure,
+					ServiceAccountName: serviceAccountName,
+					Containers: []corev1.Container{
+						{
+							Name:  "backup",
+							Image: image,
+							Command: []string{
+								"/bin/bash",
+								"-cx",
+								backupOrFixEtcd,
+							},
+							SecurityContext: &corev1.SecurityContext{
+								Privileged: to.BoolPtr(true),
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "PEER_PODS",
+									Value: peerPods,
+								},
+								{
+									Name:  "DEGRADED_NODE",
+									Value: de.Node,
+								},
+								{
+									Name:  "FIX_PEERS",
+									Value: "true",
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "host",
+									MountPath: "/host",
+									ReadOnly:  false,
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "host",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}, metav1.CreateOptions{
+		FieldManager: jobNameFixPeers,
+	})
+	if err != nil {
+		return err
+	}
+
+	timeout := adjustTimeout(ctx, log)
+	log.Infof("Waiting for %s", jobNameFixPeers)
+	select {
+	case <-time.After(timeout):
```

Comment:
@facchettos Is this in appropriate use of context to differentiate between production and testing environments?